lightly_studio/src/lightly_studio/core/dataset.py

```diff
@@ -647,16 +660,51 @@ def _generate_embeddings(session: Session, dataset_id: UUID, sample_ids: list[UU
         return
 
     embedding_manager = EmbeddingManagerProvider.get_embedding_manager()
-    model_id = embedding_manager.load_or_get_default_model(
+    model_id = embedding_manager.load_or_get_default_model(session=session, dataset_id=dataset_id)
+    if model_id is None:
+        logger.warning("No embedding model loaded. Skipping embedding generation.")
+        return
+
+    embedding_manager.embed_videos(
```

Comment:
It is still not clear to me whether this is the easiest solution.

An alternative would be:
* Add `embed_samples` to embedding_manager. It chooses the right embed function based on dataset sample type
* Keep only `_generate_embeddings` in this file

What is your opinion?